File: web/src/main/java/com/richo/reader/web/dto/Session.java
package com.richo.reader.web.dto;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.richodemus.reader.dto.UserId;

public class Session
{
	private final UserId username;
	private final String token;

	@JsonCreator
	public Session(UserId username, String token)
	{
		this.username = username;
		this.token = token;
	}

	public UserId getUsername()
	{
		return username;
	}

	public String getToken()
	{
		return token;
	}
}

File: reader/server/core/spring/src/main/kotlin/com/richodemus/reader/session.kt
import com.richodemus.reader.dto.UserId
import com.richodemus.reader.dto.Username
import javax.servlet.http.HttpSession

internal var HttpSession.userId: UserId?
    get() {
        return this.getAttribute("userId") as UserId?
    }
    set(userId) {
        this.setAttribute("userId", userId)
    }

internal var HttpSession.username: Username?
    get() {
        return this.getAttribute("username") as Username?
    }
    set(userId) {
        this.setAttribute("username", userId)
    }

internal fun HttpSession.isLoggedIn() = this.userId != null
